sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -984,7 +984,16 @@ object SparkSession extends Logging {
    *
    * @since 2.0.0
    */
+  @deprecated("This method is deprecated and will be removed in future versions.", "3.1.0")
```

Comment:
Shall we add some comments to tell users how to restore here or in the error msg?

Reply:
I agree that these APIs are not something users can call casually. 
But I am not sure the deprecated annotation can cover the behavior change here. The config is internal and will not go to the public doc. Maybe we should add a migration guide if this is not only deprecated but also desupported

Reply:
sure, i will update the migration guide as well.

Reply:
done